src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleBinaryOperator.cs

```diff
@@ -74,7 +74,21 @@ private BoundExpression ReplaceTerminalElementsWithTemps(BoundExpression expr, T
             // Examples:
             // in `expr == (..., ...)` we need to save `expr` because it's not a tuple literal
             // in `(..., expr) == (..., (..., ...))` we need to save `expr` because it is used in a simple comparison
-            return EvaluateSideEffectingArgumentToTemp(VisitExpression(expr), initEffects, ref temps);
+            BoundExpression loweredExpr = VisitExpression(expr);
+            if ((bool)(loweredExpr.Type != null) && NullableAlwaysHasValue((BoundExpression)loweredExpr) is var value && value != null)
```

Comment:
>var [](start = 108, length = 3)

We don't allow usage of var like that. I suggest declaring a local before the if and do ```(value = NullableAlwaysHasValue((BoundExpression)loweredExpr)) != null``` instead